apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+pragma solidity 0.4.15;
+
+import "./oracle/OracleInterface.sol";
+
+import "@aragon/core/contracts/apps/App.sol";
+import "@aragon/core/contracts/common/Initializable.sol";
+import "@aragon/core/contracts/common/EtherToken.sol";
+import "@aragon/core/contracts/common/erc677/ERC677Receiver.sol";
+
+import "@aragon/core/contracts/zeppelin/token/ERC20.sol";
+import "@aragon/core/contracts/zeppelin/math/SafeMath.sol";
+
+import "@aragon/apps-finance/contracts/Finance.sol";
+//import "@aragon/apps-vault/contracts/Vault.sol";
+
+
+/**
+ * @title Payroll in multiple currencies
+ * @notice For the sake of simplicity lets asume USD is a ERC20 token
+ * Also lets asume we can 100% trust the exchange rate oracle
+ */
+contract Payroll is App, Initializable, ERC677Receiver {
+    using SafeMath for uint;
+
+    bytes32 constant public OWNER_WRITE_ROLE = bytes32(1);
```

Comment:
I would rename this to 'employer role', and maybe consider splitting it in multiple roles for hiring employees, terminating, etc.